### DISCUSSION TITLE
urequests "get" does not get you anything.

### DISCUSSION THREAD

**Discussion by timonoko** [micropython/micropython#13346] [ESP32]
I really don't understand why this is so. All I want is raw HTML to parse on my own.

So I fixed it. The data was in self.text and the cryptic _parse_result() destroyed it.

Now its works exactly like  "requests.get" in real Python.

https://github.com/timonoko/CYD/blob/main/urequests_noko.py

Update: I found the problem  in "_parse_result". You cannot split('/r/n').

**Comment by karfas** [comment]
This is a problem with your data, not with the `split()` in urequests.py.

HTTP headers **MUST** be terminated by `"\r\n"`, hence the split looks correct for me.

**Reply by timonoko** [reply]
Good to know. They should fix their websites.

But in the meantime urequests_noko is the only urequests in the known universe that works perfectly with all websites, how malformed they ever may be.   

Particularly bad is micropython builtin urequests, because it can only do "OSError: [Errno 104] ECONNRESET", whatever it means.
